files/zh-cn/web/javascript/reference/operators/void/index.md

```diff
@@ -4,65 +4,91 @@ slug: Web/JavaScript/Reference/Operators/void
 ---
 {{jsSidebar("Operators")}}
 
-**`void` 运算符** 对给定的表达式进行求值，然后返回 {{jsxref("Global_Objects/undefined", "undefined")}}。
+**`void` 运算符** 对给定的表达式进行求值，然后返回 {{jsxref("undefined")}}。
+
+{{EmbedInteractiveExample("pages/js/expressions-voidoperator.html", "taller")}}
 
 ## 语法
 
-```plain
+```js-nolint
 void expression
 ```
 
 ## 描述
 
-这个运算符能向期望一个表达式的值是{{jsxref("Global_Objects/undefined", "undefined")}}的地方插入会产生副作用的表达式。
+这个运算符能向期望一个表达式的值是 {{jsxref("undefined")}} 的地方插入会产生副作用的表达式。
+
+`void` 运算符通常只用于获取 `undefined` 的原始值，一般使用 `void(0)`（等同于 `void 0`）。在上述情况中，也可以使用全局变量 {{jsxref("undefined")}} 来代替。
+
+需要注意考虑 `void` 运算符的[优先级](/zh-CN/docs/Web/JavaScript/Reference/Operators/Operator_Precedence)，以下加括号的表达式的例子可以帮助你清楚地理解 `void` 操作符的优先级：
 
-void 运算符通常只用于获取 `undefined`的原始值，一般使用`void(0)`（等同于`void 0`）。在上述情况中，也可以使用全局变量{{jsxref("Global_Objects/undefined", "undefined")}} 来代替（假定其仍是默认值）。
+```js
+void 2 === '2';   // (void 2) === '2'，返回 false
+void (2 === '2'); // void (2 === '2')，返回 undefined
+```
 
-## 立即调用的函数表达式
+## 示例
 
-在使用[立即执行的函数表达式](/zh-CN/docs/Glossary/IIFE)时，可以利用 `void` 运算符让 JavaScript 引擎把一个`function`关键字识别成函数表达式而不是函数声明（语句）。
+### 立即调用的函数表达式
+
+在使用[立即执行的函数表达式](/zh-CN/docs/Glossary/IIFE)时，`function` 关键字不可直接位于语句开头，因为会被识别成函数表达式而不是[函数声明](zh-CN/docs/Web/JavaScript/Reference/Statements/function)，并会在达到代表调用的括号时产生一个语法错误。如果是匿名函数，那么如果函数被解析为声明，就会立即产生一个语法错误。
```

Comment:
`因为会被识别成函数表达式而不是[函数声明](zh-CN/docs/Web/JavaScript/Reference/Statements/function)`

> `because that would be parsed as a [function declaration](/en-US/docs/Web/JavaScript/Reference/Statements/function)`

Reply:
```suggestion
在使用[立即调用的函数表达式](/zh-CN/docs/Glossary/IIFE)时，`function` 关键字不可直接位于语句开头，因为该表达式会被识别为[函数声明](zh-CN/docs/Web/JavaScript/Reference/Statements/function)，并会在解析到代表调用的括号时产生语法错误。如果是匿名函数，那么如果函数被解析为声明，就会立即产生语法错误。
```

Reply:
👍